options.go

```diff
@@ -53,6 +53,30 @@ func WithEndpointSettingsModifier(modifier func(settings map[string]*network.End
 	}
 }
 
+// WithEnv sets the environment variables for a container.
+// If the environment variable already exists, it will be overridden.
+// panics if the number of arguments is odd.
+func WithEnv(envs ...string) CustomizeRequestOption {
```

Comment:
To avoid panics, and let users use it without possible errors, I'd use a map of string to string instead of a variadic string. That way we enforce the contract of two strings per env var, and the logic to add them to the original `req.Env` is simpler.

Wdyt?

Reply:
I'd advocate for ease of reading instead of ease of writing, and enforcing the type is less error prone 🙏